pkg/imgpkg/cmd/copy.go

```diff
@@ -196,10 +196,25 @@ func (o *CopyOptions) writeImagesLockOutput(processedImages *ctlimgset.Processed
 		},
 	}
 
-	for _, img := range processedImages.All() {
-		imagesLock.Images = append(imagesLock.Images, lockconfig.ImageRef{
-			Image: img.DigestRef,
-		})
+	if o.LockInputFlags.LockFilePath != "" {
+		var err error
+		imagesLock, err = lockconfig.NewImagesLockFromPath(o.LockInputFlags.LockFilePath)
+		if err != nil {
+			return err
+		}
+		for i, image := range imagesLock.Images {
+			for _, img := range processedImages.All() {
+				if img.UnprocessedImageRef.DigestRef == image.Image {
```

Comment:
probably worth pulling this out as a method on processed images so it's easier to error out when there is match found (case isnt covered in this code).